### ISSUE TITLE
Add poisoning scenario metrics

### ISSUE THREAD

**Issue by reed-twosixlabs** [twosixlabs/armory#1154]
We want Armory to compute additional metrics useful for poisoning evaluations. In addition to tracking implementation considerations, this issue should also be the place to clarify what these additional metrics should be.

**Comment by reed-twosixlabs** [comment]
Initial prompts for definitions:

1. In the dirty-label threat model where we only poison training samples of a single source class relabeled as target, at inference time, we can derive any number of metrics from the confusion matrix

```
                                                  predicted source class | predicted target class | predicted other class
source class without trigger |                 
source class with trigger      |
```

Are we interested in any of these?

2. We might be interested in how the effectiveness of the poison varies as we poison different classes in a training set where the classes are unevenly distributed. (For an extreme example, if class A outnumbers class B 2:1, then you have the same amount of poison if you poison class A at 50% or class B at 100%, but then the model wouldn't know what benign instances of class B look like.) What kinds of disparities do we want to measure?

**Comment by reed-twosixlabs** [comment]
Proposals from today's meeting:

1. For filtering defenses, we can look at the class distribution of flagged data relative to the class distribution of 100% benign data (with no poison). [I can implement this.]
2. For filtering defenses, we can visualize the low-dimensional representations (e.g., t-SNE) of filtered samples, poisoned samples, etc. We can find subgroups of activation clusters and measure the distance of the filtered points from the centroid of the nearest (or some specified) cluster. [Sridevi]
2.1. Given representations of the training data, we have two ways to split: 
--- based on our majority/minority determination (TBD), and after our performers have filtered the training data,
--- filtered/unfiltered.
So we can make a 2x2 table based on where training samples live. We would hope that the decision of whether to filter is independent of majority/minority status and can test this.

**Comment by swsuggs** [comment]
Closing, addressed by #1360
